src/OpenTelemetry.Api/Trace/Tracer.cs

```diff
@@ -17,64 +17,174 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
 
 namespace OpenTelemetry.Trace
 {
     /// <summary>
     /// Tracer to record distributed tracing information.
     /// </summary>
-    public abstract class Tracer
+    public class Tracer
     {
+        private readonly ActivitySource activitySource;
+
+        internal Tracer(ActivitySource activitySource)
+        {
+            this.activitySource = activitySource;
+        }
+
         /// <summary>
         /// Gets the current span from the context.
         /// </summary>
-        public abstract TelemetrySpan CurrentSpan { get; }
+        public TelemetrySpan CurrentSpan
+        {
+            get
+            {
+                return new TelemetrySpan(Activity.Current);
+            }
+        }
 
         /// <summary>
-        /// Activates the span on the current context.
+        /// Starts span.
         /// </summary>
-        /// <param name="span">Span to associate with the current context.</param>
-        /// <param name="endSpanOnDispose">Flag indicating if span should end when scope is disposed.</param>
-        /// <returns>Disposable object to control span to current context association.</returns>
-        public abstract IDisposable WithSpan(TelemetrySpan span, bool endSpanOnDispose);
+        /// <param name="name">Span name.</param>
+        /// <param name="kind">Kind.</param>
+        /// <returns>Span instance.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public TelemetrySpan StartSpan(string name, SpanKind kind = SpanKind.Internal)
+        {
+            // TODO: Open Question - should we have both StartSpan and StartActiveSpan?
+            // Or should we call this method StartActiveSpan
+            // This method StartSpan is starting a Span and making it Active.
+            // OTel spec calls for StartSpan, and StartActiveSpan being separate.
+            // Need to see if it makes sense for .NET to strictly adhere to it.
+            // Some discussions in Spec: https://github.com/open-telemetry/opentelemetry-specification/pull/485
+            if (!this.activitySource.HasListeners())
```

Comment:
Do we need to validate that `this.activitySource` is not null?

Reply:
The only way to obtain a Tracer is via TracerProvider - which ensures activity source is not null. 
(But agree that I still need to validate inside Tracer ctor() that its given non null activitysource.)